test/e2e/helpers.go

```diff
@@ -0,0 +1,138 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	typedappsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	typedbatchv1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type deploymentsClientAdapter struct {
+	client typedappsv1.DeploymentInterface
+}
+
+func (c deploymentsClientAdapter) Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+	// TODO: honor the ctx
+	deployment, err := c.client.Get(key.Name, metav1.GetOptions{})
+	if deployObj, ok := obj.(*appsv1.Deployment); ok {
+		deployObj.Status = deployment.Status
+	}
+	return err
+}
+
+type servicesClientAdapter struct {
+	client typedcorev1.ServiceInterface
+}
+
+func (c servicesClientAdapter) Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+	// TODO: honor the ctx
+	service, err := c.client.Get(key.Name, metav1.GetOptions{})
```

Comment:
can we handle the TODO, or remove the comment if its not needed

Reply:
I removed the comment, because when CAPZ eventually moves to a newer version of client-go, this will have to change since `Get` and friends require a context as the first argument. So it should self-correct.

The alternative is to run the current `Get` in a goroutine and cancel it if the context is done. I can go that way if we prefer.

Reply:
it fine to keep it like that for now